.circleci/config.yml

```diff
@@ -488,6 +533,70 @@ jobs:
       - manage_api_cache:
           save: true
 
+  # Run Puppeteer e2e UI tests on deployed "test" environment.
+  puppeteer-e2e-test:
+    <<: *defaults
+    environment:
+      <<: *e2e_env
+      WORKBENCH_ENV: dev
+    working_directory: ~/workbench
+    parallelism: 3
+    steps:
+      - browser-tools/install-browser-tools
+      - checkout_init_git
+      - halt_job_noncode_changes
+      - run_e2e_test
+
+  # Deploy UI local server then run Puppeteer e2e tests on deployed "local" environment.
+  puppeteer-e2e-local-ui:
+    <<: *defaults
+    environment:
+      <<: *e2e_env
+      WORKBENCH_ENV: local
+    resource_class: medium+
+    working_directory: ~/workbench
+    parallelism: 1
+    steps:
+      - run:
+          name: Skip job if this is not a pull request
+          command: |
+            echo "PR: $CI_PULL_REQUEST"
+            if [ ${CIRCLE_BRANCH} != "" ] && [ ${CIRCLE_BRANCH} == "master" ] ; then
+              echo "Not a PR, skipping Puppeteer tests"
+              circleci-agent step halt
+            fi
```

Comment:
Good thinking on using master branch for skip filtering. Updated command accordingly.